pallets/xor-fee/src/lib.rs

```diff
@@ -695,7 +732,33 @@ pub mod pallet {
     pub struct Pallet<T>(PhantomData<T>);
 
     #[pallet::hooks]
-    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_idle(current_block: BlockNumberFor<T>, _remaining_weight: Weight) -> Weight {
+            let mut weight: Weight = Weight::default();
+
+            if current_block >= Self::next_update_block() {
+                // 1 read
+                match T::DynamicMultiplier::calculate_multiplier(
+                    &common::XOR.into(),
+                    &common::DAI.into(),
+                ) {
+                    Ok(new_multiplier) => {
+                        <Multiplier<T>>::put(new_multiplier); // 1 write
+                        Self::deposit_event(Event::WeightToFeeMultiplierUpdated(new_multiplier)); // 1 write
+
+                        let next_update_block = current_block + T::BlocksToUpdate::get(); // 1 read
+                        <NextUpdateBlock<T>>::put(next_update_block); // 1 write
+                        weight += T::DbWeight::get().reads(2) + T::DbWeight::get().writes(3);
+                    }
+                    Err(e) => {
```

Comment:
I'm not sure if we have to count the multiplier every block in case of an error?
Or we may set new `NextUpdateBlock`, maybe smaller than in Ok case.